src/petals/server/throughput.py

```diff
@@ -88,10 +88,30 @@ def measure_throughput_info(
 
 
 def measure_network_rps(config: BloomConfig) -> float:
-    proc = subprocess.run("python3 -m petals.cli.speed_test --json", shell=True, capture_output=True)
-    if proc.returncode != 0:
-        raise RuntimeError(f"Failed to measure network throughput (stdout: {proc.stdout}, stderr: {proc.stderr})")
-    network_info = json.loads(proc.stdout)
+    try:
+        import speedtest
+
+        s = speedtest.Speedtest()
+    except ImportError as e:
+        logger.error("Please `pip install speedtest-cli==2.1.3` or set throughput manually")
+        logger.exception(e)
+        raise e
```

Comment:
If you raise the exception, there is no need to call logger.exception, as the traceback will be printed by the raised exception

Reply:
thanks, fixed it